src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -775,13 +775,25 @@ bool UnixNativeCodeManager::GetReturnAddressHijackInfo(MethodInfo *    pMethodIn
 
     // Decode the GC info for the current method to determine its return type
     GcInfoDecoderFlags flags = DECODE_RETURN_KIND;
-#if defined(TARGET_ARM) || defined(TARGET_ARM64)
+    uint32_t codeOffset = 0;
+#if defined(TARGET_ARM)
+    PTR_UInt8 gcInfo;
+    codeOffset = GetCodeOffset(pMethodInfo, (PTR_VOID)pRegisterSet->IP, &gcInfo);
+    flags = (GcInfoDecoderFlags)(flags | DECODE_HAS_TAILCALLS | DECODE_INTERRUPTIBILITY);
+#elif defined(TARGET_ARM64)
     flags = (GcInfoDecoderFlags)(flags | DECODE_HAS_TAILCALLS);
 #endif // TARGET_ARM || TARGET_ARM64
 
-    GcInfoDecoder decoder(GCInfoToken(p), flags);
+    GcInfoDecoder decoder(GCInfoToken(p), flags, codeOffset);
     *pRetValueKind = GetGcRefKind(decoder.GetReturnKind());
 
+#if defined(TARGET_ARM)
+    // FIXME: Figure out how to encode this correctly or implement TrailingEpilogueInstructionsCount
+    if (!decoder.IsInterruptible())
+    {
+        return false;
+    }
```

Comment:
This needs to be resolved to make this PR ready. We generally have couple of options:
- Implement `TrailingEpilogueInstructionsCount` for ARM/Thumb. That's doable but error-prone.
- Teach JIT to record DWARF epilogs. Seems easy but comes at a size cost. Also, if we move to emitting ARM EHABI for exception handling, this would not work anymore since the generic EHABI codes don't cover unwinding prologs and epilogs.
- Record the size of prolog/epilog (or rather non-prolog/non-epilog ranges) in the JIT into GC info and use it here. This should have less overhead than generic DWARD unwinding and it can even be platform neutral.

Reply:
cc @VSadov

Reply:
The register corruption was a result of `INLINE_THREAD` macro trashing the `r1` registers. Once I fixed that I got several successful smoke test runs. There's still an occasional failure in the `DynamicGenerics` test. I didn't get to investigate that one since I didn't catch it with debugger yet.